copy/copy.go

```diff
@@ -241,9 +241,20 @@ func copyLayers(manifestUpdates *types.ManifestUpdateOptions, dest types.ImageDe
 		cl, ok := copiedLayers[srcLayer.Digest]
 		if !ok {
 			fmt.Fprintf(reportWriter, "Copying blob %s\n", srcLayer.Digest)
-			destInfo, diffID, err := copyLayer(dest, rawSource, srcLayer, diffIDsAreNeeded, canModifyManifest, reportWriter)
-			if err != nil {
-				return err
+			var (
+				destInfo types.BlobInfo
+				diffID   string
+				err      error
+			)
+			if !dest.SupportsForeignLayers() && len(srcLayer.URLs) != 0 {
+				destInfo = srcLayer
+				// TODO: diffID?
+				fmt.Fprintf(reportWriter, "Skipping foreign layer copy to %s\n", dest.Reference().Transport().Name())
```

Comment:
This should probably identify the blob by digest (and the `Copying Blob` message above should go into the `else` branch below).